From 0c25a850f7f336cd3bf2b0a479fe70cecb242c6e Mon Sep 17 00:00:00 2001
From: Ming Lei <ming.lei@canonical.com>
Date: Sat, 4 Aug 2012 12:01:18 +0800
Subject: [PATCH] firmware loader: remove unnecessary wmb()
Git-commit: 0c25a850f7f336cd3bf2b0a479fe70cecb242c6e
Patch-mainline: v3.7-rc1
Reference: fate#314574
Target: sle11-sp3

The wmb() inside fw_load_abort is not necessary, since
complete() and wait_on_completion() has implied one pair
of memory barrier.

Also wmb() isn't a correct usage, so just remove it.

Signed-off-by: Ming Lei <ming.lei@canonical.com>
Cc: Linus Torvalds <torvalds@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Acked-by: Lee, Chun-Yi <jlee@suse.com>

---
 drivers/base/firmware_class.c |    1 -
 1 file changed, 1 deletion(-)

--- a/drivers/base/firmware_class.c
+++ b/drivers/base/firmware_class.c
@@ -114,7 +114,6 @@ static struct firmware_priv *to_firmware
 static void fw_load_abort(struct firmware_priv *fw_priv)
 {
 	set_bit(FW_STATUS_ABORT, &fw_priv->status);
-	wmb();
 	complete(&fw_priv->completion);
 }
 
